iotdb/src/main/java/org/apache/iotdb/db/engine/overflow/io/OverflowProcessor.java

```diff
@@ -635,4 +646,37 @@ public WriteLogNode getLogNode() {
   public OverflowResource getWorkResource() {
     return workResource;
   }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
The same concern. We have to maintain these two functions if we add new fields in this class.

Reply:
See above. I suggest we make it a common practice that re-write equals() when new fields added or you are sure not overwriting it will not make any difference.